lib/spack/spack/ci.py

```diff
@@ -601,17 +623,34 @@ def generate_gitlab_ci_yaml(env, print_summary, output_file,
                         root_spec, main_phase, strip_compilers),
                     'SPACK_JOB_SPEC_PKG_NAME': release_spec.name,
                     'SPACK_COMPILER_ACTION': compiler_action,
+                    'SPACK_IS_PR_PIPELINE': str(is_pr_pipeline),
                 }
 
                 job_dependencies = []
                 if spec_label in dependencies:
-                    for dep_label in dependencies[spec_label]:
-                        dep_pkg = pkg_name_from_spec_label(dep_label)
-                        dep_spec = spec_labels[dep_label]['rootSpec'][dep_pkg]
-                        dep_job_name = get_job_name(
-                            phase_name, strip_compilers, dep_spec, osname,
-                            build_group)
-                        job_dependencies.append(dep_job_name)
+                    if enable_artifacts_buildcache:
```

Comment:
Is it possible to simplify the two branches here and below?  If not that's fine, but it seems like the logic to add dependencies is duplicated across both sides of the if statement here and in the one down further.  It seems like you could first construct a list of transitive or non-transitive dependencies based on whether the artifacts build cache is enabled, and *then* add the members of the list with a common piece of code.

Reply:
Yep, that was a little yucky.  But I think it's better now.  Let me know what you think.

Reply:
looks way better!